maven-scm-providers/maven-scm-providers-git/maven-scm-provider-gitexe/src/main/java/org/apache/maven/scm/provider/git/gitexe/command/checkin/GitCheckInCommand.java

```diff
@@ -144,8 +165,11 @@ protected CheckInScmResult executeCheckInCommand( ScmProviderRepository repo, Sc
             {
                 return new CheckInScmResult( null, statusConsumer.getChangedFiles() );
             }
-
-            Commandline clCommit = createCommitCommandLine( repository, fileSet, messageFile );
+            
+            //SCM-714: Workaround for the Windows terminal command limit
+            //
+            Commandline clCommit = createCommitCommandLine( repository, new ScmFileSet( fileSet.getBasedir() ),
```

Comment:
What is the purpose of this change? This affects `-a` to be added, but we did already add untracked changes...

Reply:
The way I read this, the line creates a commit command with files in the fileSet. However sice in the fix I already added all the file I want, I gave it the basedir, which should have no effect. Since the method is public static I didnt want to alter its behaviour so I think thats why I chose this solution. I will check the failing test during the week.